### ISSUE TITLE
CKV2_AWS_23 is triggered when route_53_record type=A is set to an elasticbeanstalk environment

### ISSUE THREAD

**Issue by frek818** [bridgecrewio/checkov#2212]
**route_53 alias records point to cloudfront distribution should not trigger CKV2_AWS_23**
The description of the CKV2_AWS_23 check is:

This check ensures that Route53 A records point to resources part of your Account rather than just random IP addresses. On the platform this check additionally compares IP's against provisioned EIP. In Checkov the graph correlates the A record against know AWS resources from EIP to Global Accelerator.

I believe that an account owned elasticbeanstalk environment should qualify as an IP address owned by the account so it should not trigger this error. Please correct me if I am wrong.

**Examples**
```hcl
resource "aws_route53_record" "site" {
  zone_id = data.aws_route53_zone.hosted_zone.id
  name    = var.sub_domain
  type    = "A"

  alias {
    name                   =  aws_elastic_beanstalk_environment.site.cname
    zone_id               =  data.aws_elastic_beanstalk_hosted_zone.current.id
    evaluate_target_health = false
  }
}
```

**Version (please complete the following information):**
 - checkov-2.0.717-py3-none-any.whl

**Comment by gruebel** [comment]
ho @frek818 I created a small test setup and the current check passes
```hcl
resource "aws_route53_record" "pass_eb" {
  zone_id = data.aws_route53_zone.dns_zone.zone_id
  name    = var.sub_domain
  type    = "A"

  alias {
    name                   =  aws_elastic_beanstalk_environment.pass_eb.cname
    zone_id                =  data.aws_elastic_beanstalk_hosted_zone.current.id
    evaluate_target_health = false
  }
}

resource "aws_elastic_beanstalk_environment" "pass_eb" {
  application = aws_elastic_beanstalk_application.example.name
  name        = "example"
}
```
are you running the check against TF files or a plan?

**Comment by frek818** [comment]
I am using the terraform plan file for evaluation. 

I don't know if this information is relevant but I'll provide incase it proves to be useful.

* The `aws_route53_record` and `aws_elastic_beanstalk_environment` resource are contained within a module.
* The terraform above is simplified. In reality there are some ternary expression that set the values. In hindsight the information might be important. The actual code looks like what's shown below. The `cloudfront_domain` variable isn't being set so it default to `null`

```hcl
variable "cloudfront_domain" {
  type = string
  default = null
}

locals {
  cloudfront_zone_id = "Z2FDTNDATAQYW2"
}

resource "aws_route53_record" "site" {
  zone_id = data.aws_route53_zone.hosted_zone.id
  name    = var.sub_domain
  type    = "A"

  alias {
    name                   =  var.cloudfront_domain != null ? var.cloudfront_domain : aws_elastic_beanstalk_environment.site.cname
    zone_id               =  var.cloudfront_domain != null ? local.cloudfront_zone_id : data.aws_elastic_beanstalk_hosted_zone.current.id
    evaluate_target_health = false
  }
}
```
